adapters/oidc/js/src/main/resources/keycloak.js

```diff
@@ -1301,6 +1304,29 @@
             return promise.promise;
         }
 
+        function checkIfRealmAvailable() {
+            return new Promise(function (resolve, reject) {
+                var url = getRealmUrl();
+                var req = new XMLHttpRequest();
+                req.open('GET', url, true);
+                req.setRequestHeader('Accept', 'application/json');
+                req.onload = function () {
+                    try {
+                        var res = JSON.parse(req.responseText);
+                        if (res.realm != kc.realm)
+                            reject(new Error('Invalid response from Keycloak realm'));
+                        resolve(res);
+                    } catch (error) {
+                        reject(error);
+                    }
+                }
+                req.onerror = function () {
+                    reject(new Error('Keycloak realm is not available'));
```

Comment:
It seems that this callback is not called when realm doesn't exist. Keycloak returns 404 response which doesn't pass `res.realm != kc.realm` check. So the error message in the catch(error) is `'Invalid response from Keycloak realm'` which doesn't seem correct.

Reply:
Yes, **onerror** handler is called only on fatal network errors. If any http response comes from server, then **onload** is called. We should do some fine grained http response code checking in **onload** handler.

Reply:
Thank you for the explanation. In that case it seems that error messages on lines 1317 and 1324 should be interchanged. Because **onerror** means there was some invalid response not that realm is not available.